src/kaczmarz/_variants.py

```diff
@@ -29,6 +32,38 @@ def _select_row_index(self, xk):
         return self._row_index
 
 
+class MaxDistanceLookahead(kaczmarz.Base):
+    """Choose equations which lead to the most progress after a 2 step lookahead
+    """
+
+    def __init__(self, *base_args, **base_kwargs):
+        super().__init__(*base_args, **base_kwargs)
+        self._next_i = None
+        self._gram = self._A @ self._A.T
+        self._gram2 = square(self._gram)
+
+    def _select_row_index(self, xk):
+        if self._next_i is not None:
+            temp = self._next_i
+            self._next_i = None
+            return temp
+
+        residual = self._b - self._A @ xk
+        residual_2 = np.square(residual)
+        cost_mat = np.array(
```

Comment:
Why do we need `np.array()` around the mess of algebra?

Reply:
Is it related to handling sparse matrices?

Reply:
Because some of the components are sparse matrices. The end result needs to be a NumPy array, so that it can be used with np.max on line 58 and for the other following operations.